test/api/profile.test.js

```diff
@@ -0,0 +1,98 @@
+import supertest from 'supertest';
+import { Storage as MockStorage } from '../../__mocks__/@google-cloud/storage';
+import { createApp } from '../../src/app';
+import { ACCEPT_VALUE_MIME } from '../../src/middlewares/versioning';
+import { generateToken, decodeToken } from '../../src/logic/jwt';
+
+beforeEach(() => MockStorage.cleanUp());
+afterEach(() => MockStorage.cleanUp());
+
+describe('DELETE /profile', () => {
+  function setup() {
+    const acceptHeader = ACCEPT_VALUE_MIME + ';version=1';
+    const agent = supertest(createApp().callback());
+    return { acceptHeader, agent };
+  }
+
+  it('gives a 200 response when successfully uploading a profile', async function() {
+    const { agent, acceptHeader } = setup();
+
+    // First send a request to create a file.
+    const result = await agent
+      .post('/compressed-store')
+      .accept(acceptHeader)
+      .type('text')
+      .send('SOME_RANDOM_CONTENT')
+      .expect(200);
+
+    if (!result.text) {
+      // Flow couldn't refine this on its own:
+      throw new Error(
+        'Could not find the text response from the post to compressed store.'
+      );
+    }
+    const jwtToken = result.text;
+    const { profileToken } = (decodeToken(jwtToken): any);
+
+    // Then delete the file.
+    await agent
+      .delete(`/profile/${profileToken}`)
+      .accept(acceptHeader)
+      .set('Authorization', `Bearer ${jwtToken}`)
+      .send()
+      .expect(200)
+      .expect('Profile successfully deleted.');
+
+    // Sending it another time will result in a 400 for it not existing.
+    await agent
+      .delete(`/profile/${profileToken}`)
+      .accept(acceptHeader)
+      .set('Authorization', `Bearer ${jwtToken}`)
+      .send()
+      .expect(400)
+      .expect('That profile was most likely already deleted.');
+  });
+
+  it('gives a 401 response when not providing a JWT', async function() {
+    const { agent, acceptHeader } = setup();
+
+    await agent
+      .delete('/profile/FAKE_HASH')
+      .accept(acceptHeader)
+      .send()
+      .expect(401)
+      .expect('Authentication Error');
+  });
+
+  it('gives a 401 response when providing an invalid JWT', async function() {
+    const { agent, acceptHeader } = setup();
+
+    await agent
+      .delete('/profile/FAKE_HASH')
+      .accept(acceptHeader)
+      .set('Authorization', `Bearer FAKE_TOKEN`)
+      .send()
+      .expect(401)
+      .expect('Authentication Error');
+  });
+
+  it('gives a 400 response when requesting to delete a profile that does not exist', async function() {
+    const { agent, acceptHeader } = setup();
+
+    // Generate a random token.
+    const profileToken = 'FAKE_HASH';
+    const jwtToken = generateToken({ profileToken });
+
+    await agent
+      .delete(`/profile/${profileToken}`)
+      .accept(acceptHeader)
+      .set('Authorization', `Bearer ${jwtToken}`)
+      .send()
+      .expect(400)
+      .expect('That profile was most likely already deleted.');
+  });
+});
```

Comment:
It would be nice to add a test about CORS, see for example what I did for the bitly patch: https://github.com/firefox-devtools/profiler-server/pull/50/commits/2957888721d6383daab9e54d5a6128876aa400d8#diff-468c685187ffe91351f1c4acdde5516bR103-R117

As you'll see in that test, there are simple utilities to check for these headers.